File: in_java/Pattern_23.java
package in_java;

public class Pattern_23 extends Pattern {
    public Pattern_23(int row, int column, String symbol) {
        super(row, column, symbol);
    }

    public void print() {
        int symbol = 1;

        for (int i = 0; i < this.row; i++) {
            symbol = 1;

            for (int j = 0; j < this.column; j++) {
                if (j <= i) {
                    System.out.print(symbol);
                    if (symbol == 1) {
                        symbol = 0;
                    } else {
                        symbol = 1;
                    }
                } else {
                    System.out.print(" ");
                }
            }
            System.out.println();
        }
    }
}

File: in_rust/src/pattern_23.rs
pub fn pattern_23() {
    let row = 6;
    let column = 6;
    let mut symbol;

    for i in 0..row {
        symbol = 1;

        for j in 0..column {
            if j <= i {
                print!("{}", symbol);
                if symbol == 0 {
                    symbol = 1;
                } else {
                    symbol = 0;
                }
            } else {
                print!(" ");
            }
        }
        println!();
    }
}
